codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Exception.java

```diff
@@ -131,6 +143,25 @@ public static int streamId(Http2Exception e) {
     }
 
     /**
+     * Used when a stream creation attempt fails but may be because the stream was previously closed.
+     */
+    public static final class ClosedStreamCreationException extends Http2Exception {
```

Comment:
Yah the name/concept is confusing but the case where this exception was used is actually a connection error.  Perhaps a new name...any suggestions :)